### ISSUE TITLE
Parameterized Runner

### ISSUE THREAD

**Issue by imminent** [robolectric/robolectric#871]
I did some tests to get one of my tests that uses `ParameterizedRobolectricTestRunner` to work without crashing. I originally was getting argument type mismatch when it tried to use the test's constructor which took an `enum` I defined (`Weekday`) and a `boolean`. I believe I have troubleshooted it down to being an issue with `enum`. This is a test that worked before with a regular `ParameterizedTestRunner`, but I later added some Android dependencies to the class, so I needed to switch to Robolectric runner. Is it possible that there's some issue with how the `enum` parameter is being used in the Robolectric test runner?

If you care, my solution was to use the int value of my enum in the `@Parameters` and convert it back to the enum in the test. And admittedly, this is an `enum` that will likely get converted to constants with static helper functions, but that's later on my To Do list.

**Comment by MGaetan89** [comment]
Any news on this? I am having the same issue...

**Comment by WonderCsabo** [comment]
@MGaetan89 you can use [my workaround](https://github.com/robolectric/robolectric/issues/871#issuecomment-42968601).

**Comment by MGaetan89** [comment]
I saw your comment, but the link is dead (I forgot to mention it in my comment ...).
I didn't try to implement it myself yet.

**Comment by markuspfeiffer** [comment]
I've just run into the same issue using Robolectric 2.4. Until this is fixed, here's a very simple workaround:

```
@ParameterizedRobolectricTestRunner.Parameters
public static Collection<Object[]> data() {
    return Arrays.asList(new Object[][] {
        {SomeEnum.VALUE_A.name(), ...},
        {SomeEnum.VALUE_B.name(), ...},
        ...
    });
}

public SomeTest(final String enumValueName, ...) {
        this.enumValue = SomeEnum.valueOf(enumValueName);
}
```

tl;dr: Use a String as the argument and then convert it to the required enum value. Its not very nice, but it works and is quite simple to do.

**Comment by WonderCsabo** [comment]
@MGaetan89 sorry, i updated the URL. My workaround works in general.

**Comment by MGaetan89** [comment]
@WonderCsabo Thanks for the update. Works find now!

**Comment by WonderCsabo** [comment]
I just updated to Robolectric 3, and i see that my [workaround](https://github.com/excilys/androidannotations/blob/develop/AndroidAnnotations/functional-test-1-5/src/test/java/org/robolectric/ParameterizedRobolectricTestRunnerWorkaround.java) for this does not work anymore, even does not compile. I checked the history of the `RobolectricTestRunner` class, and it seems it was refactored extensively. Unfortunetaly i cannot understand the changes to update my workaround :( @erd @jongerrish can i get some help on this? Ideally, we should modify the parameterized runner in Robolectric to address this bug.
